nexus/inventory/src/collector.rs

```diff
@@ -66,6 +68,7 @@ impl<'a> Collector<'a> {
 
         self.collect_all_mgs().await;
         self.collect_all_sled_agents().await;
+        self.collect_all_keepers().await;
```

Comment:
Probably, yes. Looking at the inventory collection, all of it is 100% serial. I think it all should be made concurrent. I opened an issue for this.

Reply:
There is actually a comment regarding keeping things serial: https://github.com/oxidecomputer/omicron/blob/main/nexus/inventory/src/collector.rs#L58-L63

I think instead we should probably do it in parallel and limit the concurrency with `buffer_unordered` or something.

Reply:
Yeah, limiting concurrency is a bit tricky. (You probably want all of the methods to produce iterators over futures, then operate on them via buffer-unordered. Or use an async semaphore I guess.)

I think where it would benefit us if more than one of the futures time out -- in that case (assuming the timeout is 60s) we'd spend ~60s rather than 120s+.

Reply:
> Yeah, limiting concurrency is a bit tricky. (You probably want all of the methods to produce iterators over futures, then operate on them via buffer-unordered. Or use an async semaphore I guess.)

I actually think we should do something like this. It's more complicated, but we can't be blocking inventory collections for multiple timeouts. We could easily exceed the time it takes between inventory collections.

Reply:
FWIW, we have the same problem with blueprint execution for subqueries to systems that may not be online. For` clickhouse-admin` interactions they are all in parallel, but nothing else is.